src/realizations/catchment/Bmi_Module_Formulation.cpp

```diff
@@ -195,11 +195,18 @@ namespace realization {
             throw std::runtime_error("Bmi_Singular_Formulation does not yet implement get_ts_index_for_time");
         }
 
-        const std::string Bmi_Module_Formulation::get_bmi_native_units(const std::string &name) const {
-            return get_bmi_model()->GetVarUnits(name);
+        const std::string Bmi_Module_Formulation::get_bmi_native_units(const std::string &name) {
+            // check if output is available from BMI
+            std::string bmi_var_name;
+            get_bmi_output_var_name(name, bmi_var_name);
+
+            if(!bmi_var_name.empty())
+            {
+                return get_bmi_model()->GetVarUnits(bmi_var_name);
+            }
```

Comment:
In the unwritten `else` case, there's no specified `return` at all, which is a serious bug.